core/primitives/src/types.rs

```diff
@@ -27,6 +27,15 @@ pub type Balance = u128;
 /// Gas is a type for storing amount of gas.
 pub type Gas = u64;
 
+/// Number of shards in current group.
+pub type NumShards = u64;
+/// Number of seats of validators (block producer or hidden ones) in current group.
+pub type NumSeats = u64;
+/// Number of fishermen in current group.
+pub type NumFishermen = u64;
+/// Height delta.
+pub type HeightDelta = u64;
```

Comment:
I think `HeightDelta` is confusing. `NumBlocks` might be better

Reply:
I check the code and I assume there should be both while we don't create blocks at each height.
`NumBlocks` means number of blocks in the group.
`HeightDelta` means difference of heights. It doesn't care how many blocks present in the group.

Reply:
I don't think it makes sense to introduce a type for difference between two things of some other type. Otherwise we will just keep introducing new types and actually reduce the readability of the code.

Reply:
That's a huge difference between `NumBlocks` and `HeightDelta`.
Let's say, we have an epoch from height `H` to height `(H + epoch_length)`. `epoch_length` may be equal to 100, and the actual number of blocks between `H` and `(H + epoch_length)`, let's say, 31.

Let's see what's going on in `validate_chunk_transactions()`. We call `check_blocks_on_same_chain()` and then traverse `max_difference_in_blocks` which means taking real blocks, not height difference.

Let's see how we use `epoch_length`. We don't care how many blocks we have, we just care of `height` value. Heights difference is the criteria of a new epoch.

We cannot and shouldn't mix `NumBlocks` and `HeightDelta`. They have different meaning and describe different behavior.